Scripts/Generation_scripts/Solar.py

```diff
@@ -20,14 +20,24 @@
 import json
 
 class Solar():
-    def __init__(self):
-        self.location = 'Bahraich'
-        self.CLOVER_filepath = '/***YOUR LOCAL FILE PATH***/CLOVER 4.0'
+    def __init__(self,**kwargs):
+        # kwargs must include location
+        # kwargs can include location_inputs (to override inputs in location files)
+        self.location = kwargs.get('location')
+        self.CLOVER_filepath = '.'
         self.location_filepath = self.CLOVER_filepath + '/Locations/' + self.location
         self.generation_filepath = self.location_filepath + '/Generation/PV/'
         self.input_data = pd.read_csv(self.generation_filepath + 'PV generation inputs.csv',header=None,index_col=0)[1]
         self.location_data_filepath = self.location_filepath + '/Location Data/'
-        self.location_input_data = pd.read_csv(self.location_data_filepath + 'Location inputs.csv',header=None,index_col=0)[1]
+        self.location_inputs = pd.read_csv(self.location_data_filepath + 'Location inputs.csv',header=None,index_col=0)[1]
```

Comment:
This would be a good place to use `os.path.join` as you then don't need to store `/`'s everywhere and keep track of whether you've got a `/` in place or whether you need to add one etc. 😄